test/routes/src/server/routes/t_launchpad.js

```diff
@@ -74,113 +74,163 @@ describe('The Launchpad API endpoint', () => {
       });
     });
 
-    context('when snap does not exist', () => {
-      const caveatId = 'dummy caveat';
+    context('when repo contains valid snapcraft.yaml', () => {
+      const snapName = 'dummy-test-snap';
 
       beforeEach(() => {
         nock(conf.get('GITHUB_API_ENDPOINT'))
           .get('/repos/anaccount/arepo')
           .reply(200, { permissions: { admin: true } });
         nock(conf.get('GITHUB_API_ENDPOINT'))
           .get('/repos/anaccount/arepo/contents/snapcraft.yaml')
-          .reply(200, 'name: test-snap\n');
```

Comment:
This can wait until later, but I wonder if we should have some global test setup that puts some kind of catch-all request mock in place so that requests don't slip through to production services if we forget something.